### PR TITLE
Implemented HDS Record to QDM Patient conversions

### PR DESCRIPTION
**Pull request by holmesie**
This PR includes an implementation of a module/class that converts the old HDS `Record` model to the new QDM based `Patient` model. See changes to the README.md for info on how to use it.

---

Pull requests into cqm-converter require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] Internal ticket for this PR: https://jira.mitre.org/browse/TACO-30
- [x] Internal ticket links to this PR
- [x] Code diff has been done and been reviewed
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass
- [x] If applicable, the library version number in `cqm-converter.gemspec` has been updated

**Reviewer 0:**

Name: @jbradl11 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

**Reviewer 1:**

Name: @okeefm 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

**Reviewer 2:**

Name: @ljades 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

### CODE REVIEW

File: lib/cqm/converter/hds_record.rb [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+      patient.given_names = record.first ? [record.first] : []
+      patient.family_name = record.last if record.last
+      patient.bundle_id = record.bundle_id if record.bundle_id
+
+      patient
+    end
+
+    # This helper method looks at the current state of cqm-models, and builds a hash of
+    # datatype models and their attributes.
+    def gather_qdm_model_attrs
```
Comment (okeefm) [inline lib/cqm/converter/hds_record.rb]: This code seems to also exist in `lib/cqm/converter/utils.rb`. Should it be in both places?

Comment (holmesie) [reply lib/cqm/converter/hds_record.rb]: Good catch! This is something I already fixed in the next iteration of this Gem (QDM -> HDS). At the moment, since this isn't breaking anything, I'd rather leave it as is in this PR, unless you feel strongly about it. The sooner we get this in the better IMO.

Comment (okeefm) [reply lib/cqm/converter/hds_record.rb]: Yeah, if this is fixed in new code, I'm fine leaving it for now.

**Comment by jbradl11** [comment]
Still reviewing this code, so far things look good. However, I am unable to use this as a Gem in Bonnie (issues with loading Gems and the erb file). I want to test this using the Patients in Bonnie's DB before I give it the Bonnie approval.

File: Gemfile [hunk 2]
```diff
@@ -2,6 +2,6 @@ source 'https://rubygems.org'
 
 git_source(:github) { |repo_name| "https://github.com/#{repo_name}" }
 
-gem 'cqm-models', git: 'https://github.com/projecttacoma/cqm-models.git', branch: '_type'
+gem 'cqm-models', git: 'https://github.com/projecttacoma/cqm-models.git'
```
Comment (okeefm) [inline Gemfile]: If we have the `github` git source above, do we want to use it here?

Comment (okeefm) [reply Gemfile]: Not critical to getting the PR in, but something that can get done at some point.

Comment (holmesie) [reply Gemfile]: Yes we definitely want to use the newer style, I've pushed an update.

File: lib/cqm/converter/hds_record.rb (hunk 1 above)
Comment (holmesie) [reply lib/cqm/converter/hds_record.rb]: I fixed this anyways.

**Comment by holmesie** [comment]
@jbradl11 `notes` is a Bonnie specific extension to the HDS model, which is why it is not included in these models and their converters (these converters are pure HDS). When the time eventually comes that Bonnie uses this to do a one time conversion of all Bonnie HDS Records, the Bonnie team can branch this and add any bonnie-specific fields.
